### DISCUSSION TITLE
Capture IR commands?

### DISCUSSION THREAD

**Discussion by denisidoro** [devmfc/debian-on-amlogic#63] [Ideas]
Hi! Thank you for your hard work. I've been using your builds to run Home Assistant Supervised on an H96 Max W2 box and it works great.

My question is: is there a way to capture IR (infrared) commands?

I tried following [this guide](https://wiki.archlinux.org/title/LIRC/Quick_start_guide) to no avail.

**Comment by denisidoro** [comment]
For reference, here are the "devices" Home Assistant can see:

```
1-1
1-2
autofs
cpu_dma_latency
device-mapper
event0event1
event2
event3
fb0
full
fuse
gpiochip0
hidraw0
hidraw1
hw_random
kmsg
kvm
linux,cma
loop-control
mem
mmcblk1
mmcblk1boot0
mmcblk1boot1
mmcblk1p1
mmcblk1p2
mmcblk1rpmb
null
port
random
rfkill
system
ttyS0
ttyUSB0
uhid
uleds
urandom
usb1
vfio
vga_arbiter
watchdog
watchdog0
zero
```

**Comment by denisidoro** [comment]
@devmfc could you please give me some pointers on how to make this work? I'm software engineer, so with some hints I can probably figure out the rest.

**Reply by devmfc** [reply]
Well... First of all you need a DTS node to enable an ir driver, this is for example the one for mainline S905X3:

ir: ir@8000 {
				compatible = "amlogic,meson-gxbb-ir";
				reg = <0x0 0x8000 0x0 0x20>;
				interrupts = <GIC_SPI 196 IRQ_TYPE_EDGE_RISING>;
pinctrl-0 = <&remote_input_ao_pins>;
	pinctrl-names = "default";
				status = "okay";
			};


But now, you need to find out the right settings for the S4 (look in the Vendor kernel DTS and translate that node to a mainline node, they are probably not the same). 
And you need to modify the mainline driver for 'amlogic,meson-gxbb-ir' to work for the S4, with some luck, when the hardware is compatible, this is not needed.

Long story short: look how it works in the vendor kernel (Khadas and/or Coreelec github have them)  and port that to the mainline kernel.

Good luck
